wasm/jit/jit_amd64_test.go

```diff
@@ -735,6 +747,49 @@ func TestAmd64Builder_handlePick(t *testing.T) {
 	})
 }
 
+func TestAmd64Builder_handleConstI32(t *testing.T) {
+	for _, v := range []uint32{1, 1 << 5, 1 << 31} {
+		t.Run(fmt.Sprintf("%d", v), func(t *testing.T) {
+			o := &wazeroir.OperationConstI32{Value: v}
+			builder := requireNewBuilder(t)
+			builder.initializeReservedRegisters()
+			// Dummy value not used!
+			_ = builder.locationStack.pushValueOnStack()
+
+			// Now emit the const instruction.
+			err := builder.handleConstI32(o)
+			require.NoError(t, err)
+
+			// To verify the behavior, we increment and push the const value
+			// to the stack.
+			loc := builder.locationStack.peek()
+			require.Equal(t, wazeroir.SignLessTypeI32, loc.valueType)
+			prog := builder.newProg()
+			prog.As = x86.AINCQ
+			prog.To.Type = obj.TYPE_REG
+			prog.To.Reg = loc.register
+			builder.addInstruction(prog)
+			builder.releaseRegisterToStack(loc)
+			builder.returnFunction()
+
+			// Assemble.
+			code, err := builder.assemble()
+			require.NoError(t, err)
+			// Run code.
+			eng := newEngine()
+			mem := newMemoryInst()
+			jitcall(
+				uintptr(unsafe.Pointer(&code[0])),
+				uintptr(unsafe.Pointer(eng)),
+				uintptr(unsafe.Pointer(&mem.Buffer[0])),
+			)
+			// Check the stack.
```

Comment:
adjust this and others similar to the last change you did

this was the other one.
```
			// Since we call global.get, the top of the stack must be the global value.
 			require.Equal(t, globalValue, eng.stack[0])
 			// Plus as we push the value, the stack pointer must be incremented.
 			require.Equal(t, uint64(1), eng.currentStackPointer)
```